DemoApp/Sources/Views/CallView/CallingView/DetailedCallingView.swift

```diff
@@ -8,233 +8,214 @@ import StreamVideoSwiftUI
+            let currentUser = streamVideo.user
+            let member = MemberRequest(
+                custom: currentUser.customData,
+                role: currentUser.role,
+                userId: currentUser.id
+            )
+            members.append(member)
+        }
+        return members
+    }
+
+    @State private var text = ""
+    @State private var callAction = CallAction.startCall
+    @State private var callFlow: CallFlow = .joinImmediately
+
     @State var selectedParticipants = [User]()
     @State var incomingCallInfo: IncomingCall?
     @State var logoutAlertShown = false
 
-    @ObservedObject var appState = AppState.shared
+    private var isActionDisabled: Bool {
+        guard AppEnvironment.configuration != .test else {
+            return false
+        }
+        return appState.loading || text.isEmpty
+    }
 
     var body: some View {
-        NavigationView {
-            VStack {
-                ZStack {
-                    HStack {
-                        Button {
-                            logoutAlertShown = !AppEnvironment.configuration.isRelease
-                        } label: {
-                            UserAvatar(imageURL: streamVideo.user.imageURL, size: imageSize)
-                                .accessibilityIdentifier("userAvatar")
-                        }
-                        .padding()
+        VStack {
+            DemoCallingTopView(callViewModel: viewModel)
+                .padding(.horizontal)
+                .padding(.top)
 
-                        Spacer()
-                    }
-                    Text("Call details")
-                        .font(.title)
-                        .padding()
+            HStack {
+                TextField("Call ID", text: $text)
+                    .foregroundColor(appearance.colors.text)
+                    .padding(.all, 12)
+                    .accessibilityIdentifier("callId")
+
+                Button {
+                    text = String(
+                        String
+                            .unique
+                            .replacingOccurrences(of: "-", with: "")
+                            .prefix(10)
+                    )
+                } label: {
+                    Image(systemName: "arrow.triangle.2.circlepath")
+                        .foregroundColor(.init(appearance.colors.textLowEmphasis))
+                }
+                .padding(.trailing)
+            }
+            .background(Color(appearance.colors.background))
+            .clipShape(RoundedRectangle(cornerRadius: 8))
+            .overlay(RoundedRectangle(cornerRadius: 8).stroke(Color(appearance.colors.textLowEmphasis), lineWidth: 1))
+            .padding(.bottom, 4)
+            .padding(.horizontal)
+
+            Picker("Call action", selection: $callAction) {
+                Text(CallAction.startCall.rawValue).tag(CallAction.startCall)
+                Text(CallAction.joinCall.rawValue).tag(CallAction.joinCall)
+            }
```

Comment:
When `CallAction` conforms to `CaseIterable`, then the Picker could be with `ForEach` but I guess we might not want to add all the call actions automatically. 🤔 Just a thought